rskj-core/src/main/java/org/ethereum/util/RLP.java

```diff
@@ -191,43 +195,55 @@ public static BigInteger decodeBigInteger(byte[] data, int index) {
         return BigIntegers.fromUnsignedByteArray(bytes);
     }
 
+    private static void assertLengthOfLength(byte[] payload,byte lengthOfLength) {
+        if (lengthOfLength >= payload.length) {
+            throw new RLPException("lengthOfLength cannot be higher than payload length");
+        }
+    }
+
     public static int getNextElementIndex(byte[] payload, int pos) {
 
         if (pos >= payload.length) {
-            return -1;
+            throw new RLPException("pos out of payload length range");
```

Comment:
maybe we should also check if `pos` isn't negative?

Reply:
Addressed